File: apple/Sources/ReactNativeWatchOS/Modifiers/NavigationTitle.swift
import SwiftUI

/// SwiftUI `.navigationTitle(_:)`. Apply to a view inside a
/// `NavigationStack` — typically the screen's root container — to show
/// the string as the title at the top of the watchOS navigation bar.
enum RNWNavigationTitleModifier {
    @MainActor
    static func register(into r: RNWModifierRegistry) {
        r.register("navigationTitle") { view, params, _ in
            let title = params.string("text") ?? ""
            return AnyView(view.navigationTitle(title))
        }
    }
}

File: src/modifiers/navigationTitle.ts
import { createModifier } from './createModifier';

/// SwiftUI `.navigationTitle(_:)`. Apply to a view inside a
/// `<NavigationStack>` (typically the screen's root container) to
/// display `text` as the title in the watchOS navigation bar.
export function navigationTitle(text: string) {
  return createModifier('navigationTitle', { text });
}
